### PR TITLE
Add a 'x' minute temporary lockout feature when vote kicked

### LINKED ISSUE #265: Add a 10 minute temporary ban when a player is vote kicked
Vote_kick was added as a feature that would help when no mod is on. Kicking the players is generally not the best idea as most of them might just rejoin.
Let me take an example, suppose player xyz doesn't speak English and doesn't understand it. xyz likes to grief on caverns but doesn't understand what the mods  are saying. (Future scenario) A trusted player uses vote_kick to kick him. xyz doesn't understand why he got kicked, he rejoins and starts griefing again. Now constantly kicking him would surely not resolve anything so temporarily banning him for 10 mins would help and it would at least reduce the amount of griefing done

Now that was just a basic example and it can also be used against hackers (seeing the increase in the number of hackers)

### PR DESCRIPTION
**Pull request by MinetestSam**
When a player is kicked he/she is locked out from joining until x minutes are completed
This acts as a temporary ban

All credit goes to shivajiva 

closes #265

### CODE REVIEW

**Comment by MinetestSam** [comment]
Will fix the indentation errors tomorrow

**Comment by ClobberXD** [comment]
For future PRs, keep in mind that the commit message should be in the present tense, should be brief, and should be meaningful. In this case, it could've been `Add 10 minute tempban to vote_kick` or something like that.

**Comment by MinetestSam** [comment]
The idea was to use sban first but as shivajiva ( creator of sban ) said "It would be like a nuke"

File: mods/vote/init.lua [hunk 1]
```diff
@@ -337,7 +339,12 @@ if set == nil or minetest.is_yes(set) then
 						minetest.chat_send_all("Vote passed, " ..
 								#results.yes .. " to " .. #results.no .. ", " ..
 								self.name .. " will be kicked.")
-						minetest.kick_player(self.name, "The vote to kick you passed")
+						minetest.kick_player(self.name, "The vote to kick you passed. You will be temporarily banned for 10 minutes")
```
Comment (ClobberXD) [inline mods/vote/init.lua]:
```suggestion
						minetest.kick_player(self.name, "The vote to kick you passed. You have been temporarily banned for 10 minutes")
```

Note that by the time the player sees this message, they're aleady banned.

**Comment by ClobberXD** [comment]
@Thomas--S I'd advise against including ban managers like sban into the (sub-)game. They are server-specific utilities and it's up to the server owner to include them, and choose which ban manager they want for their server.

**Comment by MinetestSam** [comment]
Tested it and it works!

**Comment by shivajiva101** [comment]
Indentation looks a bit messed up in the code below the kick.

sban is overkill but there is an api branch which implies I am going to expose some functionality like ban/unban/check ban in the future. Testing for the global sban would be a smarter way of detecting sbans presence simply because it doesn't use a minetest api function.

vote.kick_cooldown seems less convoluted and I can't see any issue exposing that as a global.

**Comment by ClobberXD** [comment]
@MinetestSam The indentation needs a little fixing like Shivajiva says. Here's how it should look like:
```
					if result == "yes" then
						minetest.chat_send_all("Vote passed, " ..
								#results.yes .. " to " .. #results.no .. ", " ..
								self.name .. " will be kicked.")
						minetest.kick_player(self.name, "The vote to kick you passed. You have been temporarily banned for 10 minutes")
						-- after player kick
						vlist[self.name].locked = true
						minetest.after(600, function()
							vlist[self.name] = nil
						end)
					else
```

**Comment by MinetestSam** [comment]
Alright done

**Comment by shivajiva101** [comment]
also the two register_on_leaveplayer functions should be combined into a single callback like this:
``` lua
minetest.register_on_leaveplayer(function(player)
        local name = player:get_player_name()
	vote.hud.players[name] = nil
	if not vlist[name].locked then
		vlist[name] = nil
	end
end)
```

**Comment by MinetestSam** [comment]
Updated!

File: mods/vote/init.lua [hunk 2]
```diff
@@ -216,8 +217,13 @@ function vote.update_hud(player)
 	end
 end
 minetest.register_on_leaveplayer(function(player)
-	vote.hud.players[player:get_player_name()] = nil
+        local name = player:get_player_name()
+		vote.hud.players[name] = nil
+		if not vlist[name].locked then
+		vlist[name] = nil
+	end
```
Comment (ClobberXD) [inline mods/vote/init.lua]: Inconsistent indentation. Should be:
```lua
	local name = player:get_player_name()
	vote.hud.players[name] = nil
	if not vlist[name].locked then
		vlist[name] = nil
	end
```

File: mods/vote/init.lua [hunk 3]
```diff
@@ -370,8 +377,7 @@ minetest.register_on_joinplayer(function(player)
         }
     end
 end)
-
-minetest.register_on_prejoinplayer(function(name, ip)
+ minetest.register_on_prejoinplayer(function(name, ip)
     if vlist[name] and vlist[name].locked then
         return "Please wait until the vote cool down period has elapsed before rejoining!"
     else
```
Comment (ClobberXD) [inline mods/vote/init.lua]: - Extra space at beginning of `minetest.register_on_prejoinplayer( ...`
- Indents are made of white-spaces. Please convert them to tab-spaces.

**Comment by MinetestSam** [comment]
Updated x2

File: mods/vote/init.lua [hunk 4]
```diff
@@ -217,11 +217,11 @@ function vote.update_hud(player)
...
-		local name = player:get_player_name()
-		vote.hud.players[name] = nil
-		if not vlist[name].locked then
-			vlist[name] = nil
-		end
+	local name = player:get_player_name()
+	vote.hud.players[name] = nil
+	if not vlist[name].locked then
+	vlist[name] = nil
+end
```
Comment (ClobberXD) [inline mods/vote/init.lua]: Oh come on, lol... I said there's one extra tab in all the lines, why did you remove two and even three tabs in the last couple of lines? :)

Comment (MinetestSam) [reply mods/vote/init.lua]: It should be fixed

File: mods/vote/init.lua (hunk 3 above)
Comment (ClobberXD) [reply mods/vote/init.lua]: Indents are still regular spaces. Please convert them to tab-spaces.

File: mods/vote/init.lua [hunk 5]
```diff
@@ -378,13 +378,13 @@ minetest.register_on_joinplayer(function(player)
...
-        for k,v in pairs(vlist) do
-            if v.ip == ip and v.locked then
-                return "You can't bypass the cool down by changing name!"
-            end
-        end
-    end
+	if vlist[name] and vlist[name].locked then
+		return "Please wait until the vote cool down period has elapsed before rejoining!"
+	else
+		for k,v in pairs(vlist) do
```
Comment (ClobberXD) [inline mods/vote/init.lua]: Space between `,` and `v` required. Sorry for being naggy, I just want to ensure that the PR adheres to the repo's code-style.

Using an IDE or a fully-featured text editor might make things much easier than to use Github's online file editor.

File: mods/vote/init.lua [hunk 6]
```diff
@@ -353,3 +367,24 @@ if set == nil or minetest.is_yes(set) then
...
+        }
+    end
+end)
+minetest.register_on_prejoinplayer(function(name, ip)
+	if vlist[name] and vlist[name].locked then
+		return "Please wait until the vote cool down period has elapsed before rejoining!"
+	else
+		for k, v in pairs(vlist) do
+			if v.ip == ip and v.locked then
+				return "You can't bypass the cool down by changing name!"
```
Comment (ClobberXD) [inline mods/vote/init.lua]: We shouldn't assume that it's the same person joining with a different name. I'd like a more generic message here.

Probably "This IP has been temporarily blocked. Please wait until the cool-down period has elapsed before rejoining."

File: mods/vote/init.lua [hunk 7]
```diff
@@ -383,7 +383,8 @@ minetest.register_on_prejoinplayer(function(name, ip)
 	else
 		for k, v in pairs(vlist) do
 			if v.ip == ip and v.locked then
-				return "You can't bypass the cool down by changing name!"
+				return "This IP has been temporarily blocked.".. 
+					"Please wait until the cool-down period has elapsed before rejoining."
```
Comment (ClobberXD) [inline mods/vote/init.lua]: When concatenated, this would result in "...blocked.Please wait...". A space is required between `.` and `Please`.

**Comment by ClobberXD** [comment]
It was agonizing to wait for 10 minutes during testing. But that got me thinking - this feature should add another chat-command for use by moderators to manually remove players from the blacklist. But I don't want to tax MinetestSam any further. I can make a PR for that later.

**Comment by MinetestSam** [comment]
You could have reduced kick_cooldown to like 60-100 so it would save your time

**Comment by ClobberXD** [comment]
> You could have reduced kick_cooldown to like 60-100 so it would save your time

I'm not that intelligent, sorry.
